File: frontend/lib/models/customer.dart
class CustomerVehicleModel {
  final int id;
  final int customerId;
  final String vehicleNo;
  final String? driverName;
  final String? notes;

  CustomerVehicleModel({
    required this.id,
    required this.customerId,
    required this.vehicleNo,
    this.driverName,
    this.notes,
  });

  factory CustomerVehicleModel.fromJson(Map<String, dynamic> json) {
    return CustomerVehicleModel(
      id: json['id'] as int,
      customerId: json['customer_id'] as int,
      vehicleNo: json['vehicle_no'] as String,
      driverName: json['driver_name'] as String?,
      notes: json['notes'] as String?,
    );
  }
}

class CustomerModel {
  final int id;
  final String accountNo;
  final String name;
  final String? phone;
  final String creditLimit;
  final String openingBalance;
  final List<CustomerVehicleModel> vehicles;

  CustomerModel({
    required this.id,
    required this.accountNo,
    required this.name,
    this.phone,
    required this.creditLimit,
    required this.openingBalance,
    required this.vehicles,
  });

  factory CustomerModel.fromJson(Map<String, dynamic> json) {
    var vList = json['vehicles'] as List? ?? [];
    return CustomerModel(
      id: json['id'] as int,
      accountNo: json['account_no'] as String,
      name: json['name'] as String,
      phone: json['phone'] as String?,
      creditLimit: json['credit_limit'].toString(),
      openingBalance: json['opening_balance'].toString(),
      vehicles: vList.map((v) => CustomerVehicleModel.fromJson(v as Map<String, dynamic>)).toList(),
    );
  }
}

class CustomerBalanceModel {
  final int customerId;
  final String accountNo;
  final String name;
  final String openingBalance;
  final String totalCreditSales;
  final String totalRecoveries;
  final String currentBalance;
  final String creditLimit;
  final String availableCredit;

  CustomerBalanceModel({
    required this.customerId,
    required this.accountNo,
    required this.name,
    required this.openingBalance,
    required this.totalCreditSales,
    required this.totalRecoveries,
    required this.currentBalance,
    required this.creditLimit,
    required this.availableCredit,
  });

  factory CustomerBalanceModel.fromJson(Map<String, dynamic> json) {
    return CustomerBalanceModel(
      customerId: json['customer_id'] as int,
      accountNo: json['account_no'] as String,
      name: json['name'] as String,
      openingBalance: json['opening_balance'].toString(),
      totalCreditSales: json['total_credit_sales'].toString(),
      totalRecoveries: json['total_recoveries'].toString(),
      currentBalance: json['current_balance'].toString(),
      creditLimit: json['credit_limit'].toString(),
      availableCredit: json['available_credit'].toString(),
    );
  }
}

File: backend/app/schemas/customer.py
from datetime import datetime
from decimal import Decimal
from typing import Optional, List
from pydantic import Field
from backend.app.schemas.common import BaseSchema

class CustomerVehicleCreate(BaseSchema):
    vehicle_no: str = Field(..., min_length=2, max_length=50)
    driver_name: Optional[str] = Field(None, max_length=100)
    notes: Optional[str] = Field(None, max_length=255)

class CustomerVehicleResponse(BaseSchema):
    id: int
    customer_id: int
    vehicle_no: str
    driver_name: Optional[str] = None
    notes: Optional[str] = None
    created_at: datetime

class CustomerCreate(BaseSchema):
    account_no: str = Field(..., min_length=3, max_length=30)
    name: str = Field(..., min_length=2, max_length=100)
    phone: Optional[str] = Field(None, max_length=20)
    credit_limit: Decimal = Field(Decimal("0.00"), ge=Decimal("0.00"))
    opening_balance: Decimal = Field(Decimal("0.00"))

class CustomerResponse(BaseSchema):
    id: int
    account_no: str
    name: str
    phone: Optional[str] = None
    credit_limit: Decimal
    opening_balance: Decimal
    created_at: datetime
    vehicles: List[CustomerVehicleResponse] = []

class CustomerBalanceResponse(BaseSchema):
    customer_id: int
    account_no: str
    name: str
    opening_balance: Decimal
    total_credit_sales: Decimal
    total_recoveries: Decimal
    current_balance: Decimal
    credit_limit: Decimal
    available_credit: Decimal
